### DISCUSSION TITLE
Ant media SDK "@antmedia/webrtc_adaptor": "^2.6.4" Unable to show Bitrate and avg bitrate while streaming and viewing live streaming

### DISCUSSION THREAD

**Discussion by yogeshJain0727** [ant-media/Ant-Media-Server#5766] [Q&A] [answered]
I want to show the average incoming bitrate and current incoming bitrate of the current live streaming. For this, I am calling the averageIncomingBitrate() and currentIncomingBitrate() functions. However, I am getting a "this.webrtcAdaptor.currentIncomingBitrate is not a function" error in our Angular (version 14) Web Applications. Can you please help me demonstrate how to call this function? Additionally, I am not receiving the "info == 'updated_stats'" response in the callback of initWebrtc. (Ant media SDK "@antmedia/webrtc_adaptor": "^2.6.4").

**Comment by lastpeony** [comment]
You should get it from server inside "info == 'updated_stats'"
Could you please share your ant media server version?

**Comment by yogeshJain0727** [comment]
Ant media server version Enterprise Edition 2.6.4 20230827_1351. but we want to get this info on my angular website in that we use "@antmedia/webrtc_adaptor": "^2.6.4"

**Reply by lastpeony** [reply]
as an alternative you can send get broadcast request to ant media server and check bitrate field of the returned object.
there is also receivedBytes
https://antmedia.io/rest/#/BroadcastRestService/getBroadcast

![image](https://github.com/ant-media/Ant-Media-Server/assets/24750845/641cc6ff-fc97-44b4-98fd-c2d9af908a01)

**Comment by yogeshJain0727** [comment]
Okay, @lastpeony, I will check if it's working for me or not.

**Comment by yogeshJain0727** [comment]
@lastpeony  I tried this API, you can see the difference between what I want to achieve(Sample page) and what I got (receivedBytes & bitrate in console) using this API in the screenshot attached below.

![image](https://github.com/ant-media/Ant-Media-Server/assets/91123237/4edfa7c0-9389-4b8c-a933-0f9e30c4c2df)



>

**Reply by lastpeony** [reply]
You should be able to get it through 			else if (info == "updated_stats") {}
of the webrtc adaptor call back.
![image](https://github.com/ant-media/Ant-Media-Server/assets/24750845/ccf2a6ca-65c8-474b-b738-51cd2a64db46)

**Comment by yogeshJain0727** [comment]
@lastpeony  I tried to call this status of the WebRTC adapter callback function but it is not returning from the server. As I already mentioned this issue at the beginning of our conversation and the version I am using.

**Reply by lastpeony** [reply]
![image](https://github.com/ant-media/Ant-Media-Server/assets/24750845/aa42fa31-63fc-431a-b24f-9c17e7435482)
the average bitrate value shown in this image is coming from  info== updated stats
if you don receive it how 485 is written there

**Reply by swatisinha-mis** [reply]
because it's a sample page not our app.

**Reply by lastpeony** [reply]
if sample on your server is working but your app is not working you need to debug your code especially the part where you init webrtc adaptor because you should be able to get the value there

**Reply by swatisinha-mis** [reply]
Can you please help us to debug, I will share you my code to you or we can connect on zoom meet also.

I am using [Github sample for publishing webrtc](https://github.com/ant-media/StreamApp/blob/master/src/main/webapp/samples/publish_webrtc.html) this code for webrtc init
Here is my mail id : swatisinhamis@gmail.com

**Reply by lastpeony** [reply]
I cannot debug your code but you can easly find where the problem is by comparing your code and the sample.

**Reply by swatisinha-mis** [reply]
I am using the same code and condition as mentioned in the sample page in my Angular application, the only difference is the sample page is using Antmedia JavaScript SDK and I am using the npm package (@antmedia/webrtc_adaptor).

**Reply by lastpeony** [reply, marked as answer]
webRTCAdaptor.enableStats(streamId);   
call this inside publish_started or play_started callbacks

**Comment by yogeshJain0727** [comment]
Thank you, @lastpeony. This works for me.
